### PR TITLE
Add trilinear_interp OP

### PR DESCRIPTION
**Pull request by heavengate**
**Add trilinear_interp OP for trilinear interpolation**
1. add trilinear_interp OP
2. add resize_trilinear python API
3. add mode 'TRILINEAR' in image_resize API

trilinear interpolation see https://en.wikipedia.org/wiki/Trilinear_interpolation

### CODE REVIEW

File: paddle/fluid/operators/interpolate_op.cu [hunk 1]
```diff
@@ -273,76 +498,148 @@ template <typename T>
...
+      int grid_dim = (pixelNum + 512 - 1) / 512;
+      grid_dim = grid_dim > 8 ? 8 : grid_dim;
+
+      if ("trilinear" == interp_method) {
+        KeTrilinearInterpBw<
+            T><<<grid_dim, 512, 0, ctx.cuda_device_context().stream()>>>(
+            input_grad_data, in_d, in_h, in_w, n, in_cdhw, output_grad_data,
+            out_d, out_h, out_w, n, out_cdhw, c, ratio_d, ratio_h, ratio_w,
+            align_corners, align_mode);
+      }
```
Comment (qingqing01) [inline paddle/fluid/operators/interpolate_op.cu]: 4D， 5D if分支和else分支代码合并下，一个函数避免过长。

Comment (heavengate) [reply paddle/fluid/operators/interpolate_op.cu]: Done，Thanks

**Review by qingqing01 — DISMISSED** [review]

**Review by wanghaoshuang — DISMISSED** [review]

File: python/paddle/fluid/layers/nn.py [hunk 2]
```diff
@@ -7821,6 +7878,112 @@ def resize_bilinear(input,
...
+    Resize input by performing trilinear interpolation based on given
+    output shape which specified by actual_shape, out_shape and scale
+    in priority order.
+
+    Trilinear interpolation is an extension of linear interpolation for 
+    interpolating functions of three variables (e.g. D-direction, 
+    H-direction and W-direction in this op) on a rectilinear 3D grid. 
+    The linear interpolation is performed on three directions.
+
+    For details of bilinear interpolation, please refer to Wikipedia:
```
Comment (guoshengCS) [inline python/paddle/fluid/layers/nn.py]: `bilinear interpolation` should be `trilinear interpolation`

Comment (heavengate) [reply python/paddle/fluid/layers/nn.py]: Done, Thanks

**Review by wanghaoshuang — APPROVED** [review]
LGTM

**Review by guoshengCS — APPROVED** [review]

**Pull request merged**